### PR TITLE
Fix issue 2963

### PR DESCRIPTION
**Pull request by michalguspiel**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [ ] 🎬 I've attached a **screen recoding** of the changes. 

## What's changed?

Describe with a few bullets **what's new:**
- Transaction model extension functions
- Report Screen and ReportScreenViewModel moves away from using legacy `Transaction`. Still 3 events are using Legacy Actions. Will change in following pull requests.
- ReportViewModel doesn't depends on `TransactionDao` directly instead it uses `TransactionRepository`
- ReportScreenState only refers to new `Transaction` model
- `SearchViewModel` doesn't depend on legacy `Transaction`
- `AccTrnsAct` utilizes Repository instead of DAO
- `CategoryIncomeWithAccountFiltersAct` uses new `Transaction` model. Legacy class added for backwards compatibility.
- `TrnsWithDateDivsAct ` uses new `Transaction` model. Legacy class added for backwards compatibility.
- `DueTrnsInfoAct` uses new `Transaction` model.
- `OverDueTrnsInfoAct` uses new `Transaction` model.
- UpcomingAct` uses new `Transaction` model.
- Refactored `ExporCSVLogic`
- Utils functions use new `Transaction` model, a lot of legacy objects added for backwards compatibility. 

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- This is a huge refactoring PR, I might have of course overlooked something. Thankfully, issue won't be closed after this PR. This is just the first step.

**In what cases your code won't work?**

- It should work, logic is unchanged

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Relates to #2963 DOES NOT CLOSE IT YET

> Replace `ISSUE_NUMBER` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Comment by michalguspiel** [comment]
@ILIYANGERMANOV  What do you think about this? Is this PR Too big? A lot of the added code is duplicated in order to keep the codebase interoperable without changing everything that uses `TransactionDao` or `legacy.Transaction`

Will create a proper PR description once this PR is submitted for the proper review.

**Comment by ILIYANGERMANOV** [comment]
> @ILIYANGERMANOV What do you think about this? Is this PR Too big? A lot of the added code is duplicated in order to keep the codebase interoperable without changing everything that uses `TransactionDao` or `legacy.Transaction`
> 
> Will create a proper PR description once this PR is submitted for the proper review.

No worries, just test it that it works 😀

File: temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/WalletValueFunctions.kt [hunk 1]
```diff
@@ -36,11 +41,94 @@ object WalletValueFunctions {
...
+
+    suspend fun transferIncome(
+        transaction: com.ivy.base.legacy.Transaction,
+        arg: Argument
     ): BigDecimal = with(transaction) {
         val condition = arg.accounts.any { it.id == this.toAccountId }
         when {
             type == TransactionType.TRANSFER && condition ->
-                exchangeInBaseCurrency(
+                LegancyExchangeTrns.exchangeInBaseCurrency(
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/WalletValueFunctions.kt:131]: Typo `Legancy`

Comment (michalguspiel) [reply temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/WalletValueFunctions.kt]: Fixed

**Comment by michalguspiel** [comment]
Ready for review, rather a big step towards moving away from `TransactionDao` and `legacy.Transaction`. More pull requests will follow to first close everything that started here and then refactor the rest of the codebase.

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**